lib/govuk_app_config/govuk_prometheus_exporter.rb

```diff
@@ -34,15 +34,19 @@ def self.configure
       end
     end
 
-    server = PrometheusExporter::Server::WebServer.new bind: "0.0.0.0", port: 9394
-    server.start
+    begin
+      server = PrometheusExporter::Server::WebServer.new bind: "0.0.0.0", port: 9394
+      server.start
 
-    if defined?(Rails)
-      Rails.application.middleware.unshift PrometheusExporter::Middleware
-    end
+      if defined?(Rails)
+        Rails.application.middleware.unshift PrometheusExporter::Middleware
+      end
 
-    if defined?(Sinatra)
-      Sinatra.use PrometheusExporter::Middleware
+      if defined?(Sinatra)
+        Sinatra.use PrometheusExporter::Middleware
+      end
+    rescue Errno::EADDRINUSE => e
```

Comment:
Would it be safer to put the rescue just after the call to `server.start` on line 39? I think as-is it'll eat exceptions from the part where it registers itself with the web framework (unlikely to throw anything, but if it does it'll be super confusing to debug!)

Reply:
Oh wait no it's no big deal cos we're only catching EADDRINUSE, duh. Really just a nit then I guess. I'm fine either way.

Reply:
Hmm, good point - It felt better to put the rescue late because if we can't bind we also want to not try to start the server, I assume (nor attach the middleware). Bear in mind that our rescue is only one specific type of error, I'd assume we would never see that specific error in the server.start call.

Reply:
Sorry for the noise